nucliadb_node/tests/common/node_writer.rs

```diff
@@ -0,0 +1,148 @@
+use nucliadb_node::writer::NodeWriterService;
+use nucliadb_node::writer::grpc_driver::NodeWriterGRPCDriver;
+use nucliadb_protos::node_writer_client::NodeWriterClient;
+use nucliadb_protos::node_writer_server::NodeWriterServer;
+use nucliadb_protos::{EmptyQuery, ShardId};
+use std::str::FromStr;
+use std::net::{SocketAddr, IpAddr};
+use tonic::Request;
+use tonic::transport::{Server, Channel};
+use tonic_health::proto::HealthCheckRequest;
+use tonic_health::proto::health_client::HealthClient;
+use tonic_health::proto::health_check_response::ServingStatus;
+
+use super::misc::linear_backoff;
+
+
+const WRITER_HOST: &str = "127.0.0.1";
+const WRITER_PORT: u16 = 18030;
+
+pub type TestNodeWriter = NodeWriterClient<Channel>;
+
+
+/// Start a Node Writer gRPC server ready to accept new connections
+pub async fn node_writer_server() {
+    tokio::spawn(async {
+        let addr = SocketAddr::new(IpAddr::from_str(WRITER_HOST).unwrap(), WRITER_PORT);
+        let writer_server = NodeWriterServer::new(
+            NodeWriterGRPCDriver::from(
+                NodeWriterService::new()
+            )
+        );
+        let (mut health_reporter, health_service) = tonic_health::server::health_reporter();
+        health_reporter
+            .set_serving::<NodeWriterServer<NodeWriterGRPCDriver>>()
+            .await;
+
+        Server::builder()
+            .add_service(health_service)
+            .add_service(writer_server)
+            .serve(addr)
+            .await
+            .expect("Error starting gRPC server")
+    });
+
+
+    let server_uri = tonic::transport::Uri::builder()
+        .scheme("http")
+        .authority(format!("{}:{}", WRITER_HOST, WRITER_PORT))
+        .path_and_query("/")
+        .build()
+        .unwrap();
+
+    let mut wait = 100;
+    const MAX_WAIT: u64 = 5000;
+    while let Err(error) = Channel::builder(server_uri.clone()).connect().await {
+        tokio::time::sleep(tokio::time::Duration::from_millis(wait)).await;
+        wait = linear_backoff(wait, 2);
+        if wait > MAX_WAIT {
+            panic!(
+                "Something went wrong while starting writer gRPC server (too many times): {:?}",
+                error
+            );
+        }
+    }
+
+    let mut stub: HealthClient<Channel> = HealthClient::new(Channel::builder(server_uri).connect().await.unwrap());
+    let result = stub.check(
+        HealthCheckRequest { service: "nodewriter.NodeWriter".to_string() }
+    ).await.unwrap();
+
+    if result.get_ref().status() != ServingStatus::Serving {
+        panic!(
+            "writer gRPC server not serving, it's {:?}!",
+            result.get_ref().status()
+        );
+    }
+}
+
+
+/// Create a new Node Writer gRPC Client
+pub async fn node_writer_client() -> TestNodeWriter {
```

Comment:
Looks like the codes from `node_writer` and the `node_reader` really similar. Maybe you can factorize it